src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj

```diff
@@ -106,16 +143,42 @@
   </ItemGroup>
   <!-- WINDOWS: CoreCLR -->
   <ItemGroup Condition=" '$(TargetsWindows)' == 'true' And '$(CoreClrOrCorRt)' == 'true'">
+    <Compile Include="System\Environment.CoreCLR.cs" />
```

Comment:
There are several different axes:

1. Classic win32 vs. UWP apps (vs. non-Windows OSes). The two Windows appmodels are very much like different OSes because of there are different APIs available and things have to be generally done differently in number of places. I believe that the most common pattern has been to use `.Win32.cs` vs. `.WinRT.cs` to provide different implementations. `System.IO.FileSystem` or  `System.Console` are good examples.

2. JIT vs. full AOT. I do not actually see any good examples. `System.Linq.Expressions` or data contract serialization where this difference is most prelevant are configured via ifdefs, not via alternative file includes.

3. Different surface of System.Private.CoreLib between CoreCLR and CoreRT/.NET Native. The common pattern for this is `.CoreCLR.cs` vs. `.netcore50.cs` or `.netcore50aot.cs`. Hopefully, most of these will go away over time as we work on filling in the missing APIs and converging the implementations.

I know there have been a ton of confusion and inconsistencies about this. We have been getting by so far because of we support just a small subset of combinations (CoreCLR/Classic, CoreCLR/UWP, CoreRT/UWP). We run into it during OSS CoreRT project bring up because it needs a new combination (CoreRT/Classic). We ended up hacking around it because of getting this cleaned up (especially wrt packaging) seemed to be a lot of work.

I am going to open an issue on fixing consistency of the .cs file names at least to make some incremental progress on this.